plugin/evm/vm.go

```diff
@@ -702,10 +702,22 @@ func (vm *VM) buildBlockWithContext(ctx context.Context, proposerVMBlockCtx *blo
 		ProposerVMBlockCtx: proposerVMBlockCtx,
 	}
 
+	currentHeadBlock := vm.blockChain.CurrentBlock()
+	orderbookTxsHeadBlockNumber := vm.txPool.GetOrderBookTxsHeadBlockNumber()
+	if orderbookTxsHeadBlockNumber > 0 && currentHeadBlock.Number.Uint64() > orderbookTxsHeadBlockNumber {
```

Comment:
i think this has to be `currentHeadBlock.Number.Uint64() >= orderbookTxsHeadBlockNumber`

Reply:
yes, nice catch. I missed that the block number being set is current head + 1. Fixed it now. Also changed the name from `OrderbookTxsHeadBlockNumber` => `orderbookTxsBlockNumber `